src/datasources/exchange-api/exchange-api.service.ts

```diff
@@ -1,51 +1,60 @@
 import { Inject, Injectable } from '@nestjs/common';
 import { ExchangeRates } from '../../domain/exchange/entities/exchange-rates.entity';
-import {
-  INetworkService,
-  NetworkService,
-} from '../network/network.service.interface';
 import { ExchangeFiatCodes } from '../../domain/exchange/entities/exchange-fiat-codes.entity';
 import { IExchangeApi } from '../../domain/interfaces/exchange-api.interface';
 import { DataSourceError } from '../../domain/errors/data-source.error';
 import { IConfigurationService } from '../../config/configuration.service.interface';
+import { CacheFirstDataSource } from '../cache/cache.first.data.source';
+import { CacheRouter } from '../cache/cache.router';
 
 @Injectable()
 export class ExchangeApi implements IExchangeApi {
   private readonly baseUrl: string;
   private readonly apiKey: string;
+  private readonly cacheTtlSeconds: number;
+
+  static readonly DEFAULT_CACHE_TTL_SECONDS = 60 * 60 * 12;
```

Comment:
Should we set an access modifier here? I mean, this should be `private` to increase encapsulation, right? 

And related to that: why do you prefer to have this parameter's default value defined within its class and the `baseUrl` in `configuration.ts`? Should we move both to the same place?

Reply:
> Should we set an access modifier here? I mean, this should be `private` to increase encapsulation, right?

Yes! 👍 

> And related to that: why do you prefer to have this parameter's default value defined within its class and the `baseUrl` in `configuration.ts`? Should we move both to the same place?

Actually I started with that solution (doing it in `configuration.ts`). But doing that there would break a lot of tests because then you need to setup the Fake Configuration in every test with this values (because it's a new value missing).

I'd actually prefer to have it in the configuration itself but that means adjusting all the tests. On the other hand we should probably start thinking about having a fake configuration by default (provided).

Reply:
Created a followup issue for this: https://github.com/safe-global/safe-client-gateway-nest/issues/318

Reply:
Great, thanks @fmrsabino 🚀